.github/workflows/main.yml

```diff
@@ -219,13 +266,22 @@ jobs:
 
       # execute
       - name: Generate PR diff
+        id: pr-diff
+        continue-on-error: true
         run: git diff origin/main...HEAD >diff.txt
       - name: Mutate Diff
+        id: mutate-diff
+        continue-on-error: true
         env:
           DISABLE_T_REX: 1
         timeout-minutes: 45
         run: cargo +nightly -Zscript scripts/mutants.rs --in-place --in-diff diff.txt
 
+      # validate
+      - name: Check for Step Failures
+        if: steps.pr-diff.outcome != 'success' || steps.mutate-diff.outcome != 'success'
+        run: exit 1
+
   coverage:
```

Comment:
These are not "checks" as such, just generating data + uploading it. If data generation fails, there is no point to try to upload (maybe even harmful). Continue on error logic not needed here.